### DISCUSSION TITLE
CreateChromatinAssay min.cells and min.features behavior?

### DISCUSSION THREAD

**Discussion by varemo** [stuart-lab/signac#902] [Q&A] [answered]
Hi,

Thanks for a great package!

I have a question about the `min.cells` and `min.features`arguments.

Let's say I have a counts matrix (feature-barcode matrix) in R with 115,468 features (rows) and 176,355 cells (columns). Let's further assume that 14 features have no counts in any cells and that 96,233 cells have zero counts. See below:

```
> dim(counts)
[1] 115468 176355

> table(rowSums(counts)==0)

 FALSE   TRUE 
115454     14

> table(colSums(counts)==0)

FALSE  TRUE 
80122 96233 
```

Running `CreateChromatinAssay` with default values for `min.features` and `min.cells` return an object with 115,468 features and 80,122 cells. That is, all features (including the 14 with no counts) are kept, whereas the 96,233 zero-count cells are removed (see below). The docs for `min.features` states "Include cells where at least this many features are detected." At least zero, would imply inclusion of cells with zero counts, at least the way I interpret this.
```
> CreateChromatinAssay(counts, fragments = frags, min.features = 0, min.cells = 0)
ChromatinAssay data with 115468 features for 80122 cells
Variable features: 0 
Genome: 
Annotation present: FALSE 
Motifs present: FALSE 
Fragment files: 1 
```

Reading the docs I would assume that setting `min.features=1` would result in including cells with at least 1 detected feature. However, with this setting only 67,155 cells are kept. These cells, in fact, have >1 counts.

```
> CreateChromatinAssay(counts, fragments = frags, min.features = 1, min.cells = 0)
ChromatinAssay data with 115468 features for 67155 cells
Variable features: 0 
Genome: 
Annotation present: FALSE 
Motifs present: FALSE 
Fragment files: 1 

> table(colSums(counts)>1)

 FALSE   TRUE 
109200  67155 
```

The `min.cells` argument works as intended, at least in the way I interpret it, as setting it to 1 keeps all features with at least 1 count (i.e. removes the 14 features with zero counts and keeping the remaining  115,454):

```
> CreateChromatinAssay(counts, fragments = frags, min.features = 1, min.cells = 1)
ChromatinAssay data with 115454 features for 67155 cells
Variable features: 0 
Genome: 
Annotation present: FALSE 
Motifs present: FALSE 
Fragment files: 1 
```

Is this behavior intended? (Maybe I am missing the definition of "detected"?) 

To me it seems that the behavior described in the docs should be:
```
counts <- counts[rowSums(counts) >= min.cells, colSums(counts) >= min.features]
```
but currently is:
```
counts <- counts[rowSums(counts) >= min.cells, colSums(counts) > min.features]
```

Anyone has any comments on this?

**Comment by timoast** [comment, marked as answer]
You're right, it should be `>= min.features` in the code. I've updated this on the develop branch.

**Reply by varemo** [reply]
Thanks Tim!
